apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c

```diff
@@ -0,0 +1,548 @@
+* @precondition     pcm should be opened before.
+* @postcondition    NA
+*/
+static void itc_audio_pcm_format_to_bits_tc_p(void)
+{
+	enum pcm_format format = 0;
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
+	format = pcm_get_format(g_pcm);
+	TC_ASSERT_NEQ_CLEANUP("pcm_get_format", format, PCM_FORMAT_NONE, pcm_close(g_pcm));
+	TC_ASSERT_NEQ_CLEANUP("pcm_format_to_bits", pcm_format_to_bits(format), 0, pcm_close(g_pcm));
+	TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_readi_p
+* @brief            read captured data from chip
+* @scenario         record voice and write data in specific file
+* @apicovered       pcm_readi
+* @precondition     pcm should be opened before
+* @postcondition    NA
+*/
+static void itc_audio_pcm_readi_tc_p(void)
+{
+	int fd = 0;
+	char input_str[INPUT_STR_LEN];
+	int ret = 0;
+	char *str = NULL;
+	char *buffer = NULL;
+	unsigned int bytes_per_frame =0;
+	int frames_read = 0;
+	int remain = 0;
+
+	buffer = malloc(pcm_get_buffer_size(g_pcm));
+	TC_ASSERT_NEQ_CLEANUP("pcm_readi", buffer, NULL, clean_all_data(0, 0));
+
+	fd = open(AUDIO_TEST_FILE, O_RDWR | O_CREAT);
+	TC_ASSERT_GT_CLEANUP("pcm_readi", fd, 0, clean_all_data(0, buffer));
+
+	bytes_per_frame = pcm_frames_to_bytes(g_pcm, 1);
+	remain = AUDIO_DEFAULT_RATE * AUDIO_RECORD_DURATION;
+
+	printf("Record will be start for 3s, press any key to start(Total frame: %d)\n", remain);
+	fflush(stdout);
+	str = gets(input_str);
+	TC_ASSERT_NEQ_CLEANUP("pcm_readi", str, NULL, clean_all_data(fd, buffer));
+
+	while (remain > 0) {
+		frames_read = pcm_readi(g_pcm, buffer, remain);
+		remain -= frames_read;
+		ret = write(fd, buffer, bytes_per_frame * frames_read);
+		TC_ASSERT_EQ_CLEANUP("pcm_readi", ret, (bytes_per_frame * frames_read), clean_all_data(fd, buffer));
+	}
+
+	printf("Record done.\n");
+	sleep(2);
+
+	clean_all_data(fd, buffer);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_write_p
+* @brief            play captured data from filesystem
+* @scenario         play recored file previously
+* @apicovered       pcm_writei
+* @precondition     pcm_readi should be opened before
+* @postcondition    NA
+*/
+static void itc_audio_pcm_writei_tc_p(void)
+{
+	int fd = 0;
+	char *buffer = NULL;
+	ssize_t size = 0;
+	ssize_t bytes = 0;
+	int ret = 0;
+	unsigned int total_frames_write = 0;
+	unsigned int rate = 0;
+	int play = 1;
+
+	fd = open(AUDIO_TEST_FILE, O_RDONLY);
+	TC_ASSERT_GEQ("pcm_writei", fd, 0);
+
+	/* use default config here */
+	g_pcm = pcm_open(0, 0, PCM_OUT, NULL);
+	TC_ASSERT_GT_CLEANUP("pcm_writei", pcm_is_ready(g_pcm), 0, close(fd));
+
+	size = pcm_get_buffer_size(g_pcm);
+	rate = pcm_get_rate(g_pcm);
+	bytes = rate * (pcm_format_to_bits(pcm_get_format(g_pcm)) / 8) * pcm_get_channels(g_pcm) * AUDIO_RECORD_DURATION;
+
+	buffer = (char *)malloc(bytes);	//rate * bit * channels * duration
```

Comment:
please check utc this will takes huge size buffer.

Reply:
@Taejun-Kwon  Updated testcase as per request. Please check